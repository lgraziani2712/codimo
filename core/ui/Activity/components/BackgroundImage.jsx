/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import CursorURL from '../images/cursor.png';
import PointerURL from '../images/pointer.png';

type Props = {|
  children?: React.Element<*>,
  image: ?string,
|};
const blurImage = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+' +
                  'AAAAG0lEQVQIW2NkYGD4z8DAwMgABXAGNgGwSgwVAFbmAgXxvZSoAAAAAElFTkSuQmCC';
const CenterGame = styled.div`
  align-items: center;
  cursor: url(${CursorURL}) 4 0, auto;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;

  &::before {
    background: ${({ image }: Props) => (image ? `url(${image}) repeat` : 'inherit')};
    content: '';
    filter: contrast(65%) brightness(110%) saturate(75%) sepia(22%) grayscale(20%);
    height: 100%;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }

  &::after {
    background: url(${blurImage}) repeat;
    content: '';
    height: 100%;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }

  & a,
  & button {
    cursor: url(${PointerURL}) 15 0, auto;
  }
`;

const BackgroundImage = ({ children, image }: Props) => (
  <CenterGame image={image}>
    {children}
  </CenterGame>
);

export default BackgroundImage;
