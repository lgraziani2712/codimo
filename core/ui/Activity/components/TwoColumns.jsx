/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import styled from 'styled-components';

const TwoColumns = styled.div`
  align-items: flex-end;
  display: flex;
  filter: drop-shadow(2px 2px 8px rgba(0, 0, 0, 0.8));
`;

export default TwoColumns;
