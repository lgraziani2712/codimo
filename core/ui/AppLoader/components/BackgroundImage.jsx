/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

type Props = {|
  children?: React.Element<*>,
  image: string,
|};
const blurImage = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+' +
                  'AAAAG0lEQVQIW2NkYGD4z8DAwMgABXAGNgGwSgwVAFbmAgXxvZSoAAAAAElFTkSuQmCC';
const CenterGame = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
  &::before {
    background: url(${({ image }: Props) => image}) repeat;
    content: '';
    filter: contrast(65%) brightness(110%) saturate(75%) sepia(22%) grayscale(20%);
    height: 100%;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }
  &::after {
    background: url(${blurImage}) repeat;
    content: '';
    height: 100%;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }
`;

const BackgroundImage = ({ children, image }: Props) => (
  <CenterGame image={image}>
    {children}
  </CenterGame>
);

export default BackgroundImage;
