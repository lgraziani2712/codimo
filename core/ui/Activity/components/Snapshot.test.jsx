/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

import BackgroundImage from './BackgroundImage';
import TwoColumns from './TwoColumns';

describe('ui/Activity/components', () => {
  it('renders <BackgroundImage /> correctly', () => {
    const buttonTree = renderer.create(
      <BackgroundImage image="/test.png" />,
    ).toJSON();

    expect(buttonTree).toMatchSnapshot();
  });
  it('renders <TwoColumns /> correctly', () => {
    const tree = renderer.create(<TwoColumns />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
