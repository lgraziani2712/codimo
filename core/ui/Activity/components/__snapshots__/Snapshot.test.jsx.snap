// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Activity/components renders <BackgroundImage /> correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: url(cursor.png) 4 0,auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c0::before {
  background: url(/test.png) repeat;
  content: '';
  -webkit-filter: contrast(65%) brightness(110%) saturate(75%) sepia(22%) grayscale(20%);
  filter: contrast(65%) brightness(110%) saturate(75%) sepia(22%) grayscale(20%);
  height: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: -1;
}

.c0::after {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAAG0lEQVQIW2NkYGD4z8DAwMgABXAGNgGwSgwVAFbmAgXxvZSoAAAAAElFTkSuQmCC) repeat;
  content: '';
  height: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: -1;
}

.c0 a,
.c0 button {
  cursor: url(pointer.png) 15 0,auto;
}

<div
  className="c0"
/>
`;

exports[`ui/Activity/components renders <TwoColumns /> correctly 1`] = `
.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-filter: drop-shadow(2px 2px 8px rgba(0,0,0,0.8));
  filter: drop-shadow(2px 2px 8px rgba(0,0,0,0.8));
}

<div
  className="c0"
/>
`;
